javascript/code-challenges/challenges-04.test.js

```diff
@@ -0,0 +1,379 @@
+CHALLENGE 3
+
+Write a function called sortNumbers that takes an array of numbers and sorts them from smallest to largest.
+
+HINT: Beware... JS default is "Lexical" ordering.
+------------------------------------------------------------------------------------------------ */
+
+const sortNumbers = (arr) => {
+  let temp = arr.sort( (a, b) => {
+    return (a - b);});
+  return temp;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named sortBackwards that takes in an array of numbers and returns the same array, with the numbers sorted, largest to smallest.
+
+HINT: Do it with a custom sort callback, not with using `.reverse()`. ;)
+------------------------------------------------------------------------------------------------ */
+
+const sortBackwards = (arr) => {
+  arr = arr.sort( (a, b) => {
+    return (b - a);});
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named alphabetize that takes in an array of strings and returns the same array with the strings sorted alphabetically.
+
+In this alphabetization, capital letters come before lower case letters.
+
+For example, ['Alphabet', 'Zebra', 'alphabet', 'carrot'] is correctly sorted.
+------------------------------------------------------------------------------------------------ */
+
+const alphabetize = (arr) => {
+  arr = arr.sort();
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named sortByPrice that takes in an array of objects, each of which has a 'price' property, and sorts those objects by price, lowest to highest, returning the same array.
+
+Here is an example of the input:
+[
+  {name: 'Sweatshirt', price: 45},
+  {name: 'Bookmark', price: 2.50},
+  {name: 'Tote bag', price: 15}
+];
+------------------------------------------------------------------------------------------------ */
+
+const sortByPrice = (arr) => {
+  arr = arr.sort(function (a, b) {
+    return a.price - b.price;});
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 7 - Stretch Goal
+
+Write a function named alphabetizeBetter that takes in an array of strings and returns the same array, with the strings sorted alphabetically. Capitalization should not change the sort order of two strings.
+
+For example, ['Alphabet', 'alphabet', 'carrot', 'Zebra'] is correctly sorted, and so is ['alphabet', 'Alphabet', 'carrot', 'Zebra'].
+------------------------------------------------------------------------------------------------ */
+
+const alphabetizeBetter = (arr) => {
+  let temp = ((arr.sort((a,b)=> a.toUpperCase() > b.toUpperCase() ? 1 : -1)));
+  return temp;
+};
+
```

Comment:
So this is one solution and it totally works. Another way I've seen it done was using a string prototype .localeCompare().
It compares two things to see if they are equivalent. So it will take a lower or uppercase, compare them and note that "A" and "a" are the "same", and therefore sort "a" before "C", even though the unicode for "C" is of a lower value than the unicode for "a". If that makes any sense at all.